src/context/productListContext.js

```diff
@@ -1,21 +1,49 @@
-import { createContext, useContext, useReducer} from "react";
-import {productReducer} from "../reducer/productList/productListReducer"
+import { createContext, useContext, useReducer } from "react";
+import { productReducer } from "../reducer/productList/productListReducer";
+import {
+  sortByPrice,
+  filterByRating,
+  filterByCategory,
+  filterByPriceRange,
+} from "../reducer/productList/helper";
 
-const ProductListContext = createContext()
-const useProductList = () => useContext(ProductListContext)
+const ProductListContext = createContext();
+const useProductList = () => useContext(ProductListContext);
 
-const initialValue = {
+const ProductProvider = ({ children }) => {
+  const initialValue = {
     loading: false,
-    productListData: []
-}
+    productListData: [],
+    sortDir: "",
+    ratingAmount: "",
+    priceRange: "",
+    category: {
+      fashion: false,
+      jewellery: false,
+      home: false,
+    },
+    // category: {},
+  };
 
-const ProductProvider = ({children}) => {
-    const [state, dispatch] = useReducer(productReducer, initialValue)
-    return (
-        <ProductListContext.Provider value={{state,dispatch}}>
-            {children}
-        </ProductListContext.Provider>
+  const [state, dispatch] = useReducer(productReducer, initialValue);
+
+  const filteredData = filterByPriceRange(
+    state,
+    filterByCategory(
+      state,
+      filterByRating(state, sortByPrice(state, state.productListData))
     )
-}
+  );
+
+  const composeFunc = (state, func) => {
+    func.reduce((acc, curr) => curr(acc), state);
+  };
```

Comment:
you have created a composeFunc but you are not using it, and compose function is supposed to return a function, I think you can improve  this  instead of calling different filter functions one by one